frontend/src/components/pages/profile/Profile.tsx

```diff
@@ -1,10 +1,186 @@
+   */
+  const serviceHandlers = React.useContext(ServiceContext);
+  const profileHandlerService = serviceHandlers.profileHandlerService;
+  
+  const [person, setPerson] = React.useState<PersonProps|undefined>(undefined);
+  const [profileId, setProfileId] = React.useState<string>(undefined);
+  const [submitSuccess, setSubmitSuccess] = React.useState<Boolean|undefined>(undefined);
+  
+  React.useEffect(() => {
+    (async() => {
+      const personPromise = await profileHandlerService.getPerson(USER_ID);
+      setPerson(personPromise);
+    })();
+  }, [profileId]); /** Re-renders Profile only when the userId changes */
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, email: event.target.value});
+  }
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, nickname: event.target.value});
+  }
+
+  const handlePronounsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, pronouns: event.target.value});
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    (async() => {
+      const success = (person.email === "" || person.nickname === "")
+        ? false /** At least one required field is missing */
+        : await profileHandlerService.updatePerson(person);
+      setSubmitSuccess(success);
+    })();
+  };
 
-export const Profile = () => {
   return (
-    <div>
-      <h2>Profile</h2>
-      <p>This is the Profile page.</p>
+    <div className={classes.root}>
+      <form>
+        <div>
+          <TextField
+            fullWidth
+            helperText="to be displayed around CoffeeHouse"
+            id="nickname"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Nickname"
+            margin="normal"
+            onChange={handleNameChange}
+            placeholder="Nickname"
+            style={{ margin: 8 }}
+            value={person? person.nickname : ""}
+            variant="outlined"
+          />
+          <TextField
+            fullWidth
+            helperText="to be displayed around CoffeeHouse"
+            id="pronouns"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Pronouns"
+            margin="normal"
+            onChange={handlePronounsChange}
+            placeholder="Pronouns"
+            style={{ margin: 8 }}
+            value={person? person.pronouns : ""}
+            variant="outlined"
+          />
+          <TextField
+            fullWidth
+            id="email"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Email"
+            margin="normal"
+            onChange={handleEmailChange}
+            placeholder="Email"
+            style={{ margin: 8 }}
+            value={person? person.email : ""}      
+            variant="outlined"
+          />
```

Comment:
Seems as though there's a lot of duplication here, is there any way to maybe map this format over the data?

Reply:
I think we should absolutely look into this, but my only hesitation is because not all of the TextFields have the same attributes, I think there could be some pretty messy statements. With some work I believe I could get it down, but the variation would still make it somewhat messy.